core/hooks/contracts/useContractEvents.ts

```diff
@@ -7,36 +6,35 @@ import {
   ContractEventsContext,
 } from '../../providers/contractEvents/mod.ts';
 import { getExpiredItem } from '../../utils/getExpiredItem.ts';
-import { useApi } from '../useApi.ts';
 import { useBlockHeader } from '../substrate/useBlockHeader.ts';
 import { useConfig } from '../useConfig.ts';
 import { useInterval } from '../useInterval.ts';
+import { ChainContract } from './useContract.ts';
 
 export const useContractEvents = (
-  address: string,
-  abi: Abi,
+  chainContract?: ChainContract,
 ): ContractEvent[] => {
   const { events, addContractEvent, removeContractEvent } = useContext(
     ContractEventsContext,
   );
-  const { api } = useApi();
-  const { blockNumber, header } = useBlockHeader();
+  const { blockNumber, header } = useBlockHeader(chainContract?.chainId) || {};
   const C = useConfig();
 
-  const eventsForAddress = events[address] || [];
+  const address = chainContract?.contract?.address.toString() || '';
```

Comment:
`address` can be `undefined`? if so 👇 

```suggestion
  const address = chainContract?.contract?.address?.toString() || '';
```